pkg/kubectl/cmd/update.go

```diff
@@ -88,27 +104,27 @@ func updateWithPatch(cmd *cobra.Command, args []string, f *Factory, patch string
 	return name
 }
 
-func updateWithFile(cmd *cobra.Command, f *Factory, filename string) string {
+func updateWithFiles(cmd *cobra.Command, f *Factory, info *resource.Info) error {
 	schema, err := f.Validator(cmd)
-	checkErr(err)
-	mapper, typer := f.Object(cmd)
-
-	clientConfig, err := f.ClientConfig(cmd)
-	checkErr(err)
-	cmdApiVersion := clientConfig.Version
-
-	mapping, namespace, name, data := ResourceFromFile(filename, typer, mapper, schema, cmdApiVersion)
+	if err != nil {
+		return err
+	}
 
-	client, err := f.RESTClient(cmd, mapping)
-	checkErr(err)
+	data, err := info.Mapping.Codec.Encode(info.Object)
+	if err != nil {
+		return err
+	}
+	if err := schema.ValidateBytes(data); err != nil {
+		return err
+	}
 
 	cmdNamespace, err := f.DefaultNamespace(cmd)
```

Comment:
Create factored this differently. Functions dependent just on cmd, such as Validator() and DefaultNamespace() were called in Run.